### ISSUE TITLE
[UX] No UI feedback when prompting password

### ISSUE THREAD

**Issue by desbma** [pimalaya/neverest#17]
When using `right.backend.auth.cmd` to get password from a program that itselfs prompts a password (ie. `keepassxc-cli` to unlock a database), there is absolutely no UI feedback, so `neverest` appears to be frozen, when the subprocess is just waiting for a password input.

Setting `right.backend.auth.cmd = "echo 'Enter password> ' && keepassxc-cli ...` as a workaround does not work either, because the process stdout is captured by neverest.

**Comment by soywod** [comment]
> there is absolutely no UI feedback

Indeed, stdin is not forwarded when spawning commands. Commands are responsible for prompting their own input. For example, when running GPG commands, people should use the right flavour of [pinentry](https://www.gnupg.org/related_software/pinentry/index.html) (GTK, QT…).

That said, Neverest could check if the current terminal is a TTY:

- If so, then it can forward stdin to send input to commands
- If not, keep the actual behaviour (do not forward stdin)

**Comment by desbma** [comment]
> That said, Neverest could check if the current terminal is a TTY:
> 
>     * If so, then it can forward stdin to send input to commands
> 
>     * If not, keep the actual behaviour (do not forward stdin)

Sounds good to me.

Another nice addition would be to output a message like `Getting password with command XXX...`, that way the user would now Neverest is waiting for a subprocess. This is the root issue in my opinion, when I added `--debug`, I immediately understood what was going on.

**Comment by soywod** [comment]
> Another nice addition would be to output a message like `Getting password with command XXX...`, that way the user would now Neverest is waiting for a subprocess. This is the root issue in my opinion, when I added `--debug`, I immediately understood what was going on.

I am not convinced about a message, I believe logs are fine. They should be used exactly the way you did: when you have a problem and try to dig. Otherwise I am afraid such messages will disturb the whole experience.

Let's keep this issue open for the stdin forwarding. Thank you for your feedback!

**Comment by soywod** [comment]
I'm afraid this feature is not doable. I understand better now this sentence:

> Another nice addition would be to output a message like `Getting password with command XXX...`, that way the user would now Neverest is waiting for a subprocess.

In fact the best would be to forward stdin AND stdout, so that users get full control. But how to distinguish a command that expects stdin/out from one that does not? We cannot know if the stdout content is a prompt or the actual password.

For this reason I believe the feature cannot be properly implemented, and is too risky to attempt. It is better, IMO, to let users deal with prompting in their command. I personally use `pass`, which uses `gpg`. I have a GPG agent that runs in background, and uses `pintentry` to display the right prompt for my environment. In my case it shows me a GTK-based prompt that I can see everywhere, it's not tied up to any TTY.
